### ISSUE TITLE
Incorrect bounding rect returned from pygame.draw.arc when width>=0

### ISSUE THREAD

**Issue by thiagojobson** [pygame/pygame#1212]
I think it might have an inconsistency regarding the docs and the actual behavior of `draw.arc` when width param is 0.

**Expected result**: Nothing is drawn and returned bounding rect width and height are set to zero.

**Actual result**: Nothing is drawn but returned bounding rect is the same as the `rect` param.

According to docs:
> if nothing is drawn the bounding rect's position will be the position of the given rect parameter and its width and height will be 0

This script shows that even when width = 0 (nothing is drawn), the returned bounding rect's width and height are not 0.
 

```
import time
import pygame
from math import radians

surf = pygame.display.set_mode((200, 200))

r = pygame.Rect(0, 0, 10, 10)
br = pygame.draw.arc(surf, (255, 0, 255), r, radians(0), radians(90), width=0)
print('bouding rect = {} | rect = {}'.format(br, r))

pygame.display.flip()

time.sleep(2)
pygame.quit()
```

Output:
```
pygame 2.0.0.dev3 (SDL 2.0.9, python 3.7.3)
Hello from the pygame community. https://www.pygame.org/contribute.html
bouding rect = <rect(0, 0, 10, 10)> | rect = <rect(0, 0, 10, 10)>

```
System details:
 - Lubuntu (64 bits)
 - Python: 3.7.3 (64 bits)
 - pygame: 2.0.0.dev3 (SDL: 2.0.9)

**Comment by charlesej** [comment]
The bounding rects for widths > 0 are also incorrect in some instances. The following code gives an example of an incorrect bounding rect.

```python
import math
import pygame


def create_bounding_rect(surface, start, surf_color):
    """Create a rect to bound all the pixels that don't match surf_color."""
    width, height = surface.get_clip().size
    xmin, ymin = width, height
    xmax, ymax = -1, -1

    for y in range(height):
        for x in range(width):
            if surface.get_at((x, y)) != surf_color:
                xmin = min(x, xmin)
                xmax = max(x, xmax)
                ymin = min(y, ymin)
                ymax = max(y, ymax)

    if -1 == xmax:
        # No points means 0 sized rect with the position at the start.
        return pygame.Rect(start, (0, 0))

    return pygame.Rect((xmin, ymin), (xmax - xmin + 1, ymax - ymin + 1))


surf_color = pygame.Color('black')
surface = pygame.display.set_mode((300, 300))
surface.fill(surf_color)
rect = pygame.Rect((0, 0), (50, 50))
rect.center = surface.get_rect().center

bounding_rect = pygame.draw.arc(surface, pygame.Color('red'), rect, 0,
                                math.pi, 1)
actual_rect = create_bounding_rect(surface, rect.topleft, surf_color)

print('{} reported bounding rect'.format(bounding_rect))
print('{} actual bounding rect'.format(actual_rect))
```

The above code gives the following output showing the actual bounding rect is different than the bounding rect returned from `pygame.draw.arc`. The arc being drawn is not what is being reported as drawn.
```
pygame 2.0.0.dev3 (SDL 2.0.10, python 3.7.4)
Hello from the pygame community. https://www.pygame.org/contribute.html
<rect(125, 125, 50, 50)> reported bounding rect
<rect(125, 125, 51, 26)> actual bounding rect
```

System details:
- os: windows 10 (64bit)
- python: 3.7.4 (64bit)
- pygame: 2.0.0.dev3 (SDL: 2.0.10) at 7eabb6ff3273a927ae544916ccf25004e5077b12

**Comment by charlesej** [comment]
A test for widths > 0 was added with an `@unittest.expectedFailure` decorator via #1309.

**Comment by charlesej** [comment]
Resolved by #1494.

**Comment by MightyJosip** [comment]
@charlesej When you have the time and when we finish critical issues can you maybe edit test_arc__bounding_rect() because it checks for a big number of cases (76464) and that lasts around 10x longer than other tests for bounding rect (22 sec on my PC).
